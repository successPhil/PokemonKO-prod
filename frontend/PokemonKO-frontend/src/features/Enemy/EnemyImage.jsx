import { motion } from "framer-motion"
import TrainerContext from "../../contexts/TrainerContext"
import { useContext } from "react"
export default function EnemyImage({enemyImage}){
    const { animateEnemy } = useContext(TrainerContext)
    // console.log(animateEnemy, 'INSIDE ENEMY IMAGE COMPONENT')
    return (
    <motion.div 
        initial={{x:0, y:0}}
        animate={animateEnemy ? { x: -200, y: 130, rotate: -30 } : { x: 0, y: 0, rotate: 0 }}
        transition={{ duration: 0.4 }} // Adjust the duration as needed 
        className="enemy-image">      
    <img src={enemyImage} style={{
      height: '420px',
      width: '100%', // This will make the image take up the full width of its container
      objectFit: 'contain', // This ensures the image is contained within the specified height and width
    }}
    />
    </motion.div>)

}