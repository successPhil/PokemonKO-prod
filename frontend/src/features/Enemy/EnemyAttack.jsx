import { motion } from "framer-motion"
import TrainerContext from "../../contexts/TrainerContext"
import { useContext } from "react"

export default function EnemyAttack() {
    const { animateEnemyAttack , animateColor, selectPokemon} = useContext(TrainerContext)
    let pokeType = ""

    if (selectPokemon){
        pokeType = selectPokemon.types.split(", ")[0]
    }

    return (
        <motion.div
        className="enemy-attack"
        initial={{x:1000, y:170}}
        animate={animateEnemyAttack ? { x: 750, y: 260, rotate: 0 } : { x: 1000, y: 170, rotate: 0 }}
        transition={{ duration: 0.4 }}
        >
            <div>
        <span className={`pokeball-attack ${animateColor}`}>Z</span>
        <span className={`pokeball-attack ${pokeType}`} >u</span>
      </div>
        </motion.div>
    )
}