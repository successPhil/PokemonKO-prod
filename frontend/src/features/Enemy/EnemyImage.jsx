import { motion } from "framer-motion"
import TrainerContext from "../../contexts/TrainerContext"
import { useContext } from "react"
export default function EnemyImage({enemyImage, isMobile}){
    const { animateEnemy } = useContext(TrainerContext)
    // console.log(animateEnemy, 'INSIDE ENEMY IMAGE COMPONENT')
    const mobileImage = isMobile ? "mobile-enemy-image" : "enemy-image"
    return (
    <motion.div 
        initial={{x:0, y:0}}
        animate={animateEnemy ? { x: -200, y: 130, rotate: -30 } : { x: 0, y: 0, rotate: 0 }}
        transition={{ duration: 0.4 }} // Adjust the duration as needed 
        className={mobileImage}>      
    <img src={enemyImage} style={{
      height: '420px',
      width: '100%', // This will make the image take up the full width of its container
      objectFit: 'contain', // This ensures the image is contained within the specified height and width
    }}
    />
    </motion.div>)

}