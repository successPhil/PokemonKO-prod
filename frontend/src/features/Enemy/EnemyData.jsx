import GameText from "../../styles/PokemonGB"
import ProgressBar from "../ProgressBar/ProgressBar"
import PlayerBar from "../Player/PlayerBar"
import TrainerContext from "../../contexts/TrainerContext"
import { useContext } from "react"
import EnemyImage from "./EnemyImage"
import EnemyAttack from "./EnemyAttack"

export function capitalizeFirst(name){
    return name[0].toUpperCase() + name.slice(1)

}

export default function EnemyData() {
    const { enemyPokemon, animateEnemyAttack } = useContext(TrainerContext)

    if (enemyPokemon){
        return (<>
        <EnemyImage enemyImage={enemyPokemon.front_image_url}/>
        {animateEnemyAttack && <EnemyAttack />}
        <PlayerBar /> 
        <div className="enemy-name">
            <GameText>{capitalizeFirst(enemyPokemon.name)}</GameText>
        </div>
        <div className="enemy-level">
            <GameText>LVL:{enemyPokemon.level}</GameText>
        </div>
        <div className="enemy-health-bar">
        <GameText>HP:</GameText>
        <ProgressBar value={enemyPokemon.health} maxValue={enemyPokemon.max_health} barColor={'#DA2C38'} />
        </div>
        </> )
    }
    }
