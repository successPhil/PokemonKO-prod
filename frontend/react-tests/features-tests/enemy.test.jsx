import EnemyAttack from '../../src/features/Enemy/EnemyAttack'
import EnemyData from '../../src/features/Enemy/EnemyData'
import EnemyImage from '../../src/features/Enemy/EnemyImage'
import {expect, test} from 'vitest'

test("Enemy attack, data, and image renders without crashing", ()=> {
    const enemyAttack = <EnemyAttack/>
    expect(enemyAttack).toMatchSnapshot()
    const enemyData = <EnemyData/>
    expect(enemyData).toMatchSnapshot()
    const enemyImage = <EnemyImage/>
    expect(enemyImage).toMatchSnapshot()
})