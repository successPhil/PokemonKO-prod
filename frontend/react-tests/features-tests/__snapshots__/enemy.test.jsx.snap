// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Enemy attack, data, and image renders without crashing 1`] = `<EnemyAttack />`;

exports[`Enemy attack, data, and image renders without crashing 2`] = `<EnemyData />`;

exports[`Enemy attack, data, and image renders without crashing 3`] = `<EnemyImage />`;
